src/debug/gdbstub.rs

```diff
@@ -572,21 +567,11 @@ pub mod svsm_gdbstub {
             // Get the current task from the stopped CPU so we can mark it as stopped
             let tl = TASKLIST.lock();
             let str = match tl.get_task(tid.get() as u32) {
-                Some(t) => {
-                    let t = t.task.lock_read();
-                    match t.state {
-                        TaskState::RUNNING => {
-                            if let Some(allocation) = t.allocation {
-                                if this_cpu().get_apic_id() == allocation {
-                                    "Stopped".as_bytes()
-                                } else {
-                                    "Running".as_bytes()
-                                }
-                            } else {
-                                "Stopped".as_bytes()
-                            }
-                        }
-                        TaskState::TERMINATED => "Terminated".as_bytes(),
+                Some(task) => {
+                    if task.is_running() {
+                        "Running".as_bytes()
+                    } else {
+                        "Terminated".as_bytes()
```

Comment:
Now the idle tasks appear in the task list it has highlighted one last issue where blocked tasks are marked as terminated:

```suggestion
                Some(task) => {
                    if task.is_running() {
                        "Running".as_bytes()
                    } else {
                        "Terminated".as_bytes()                Some(task) => {
                    if task.is_running() {
                        "Running".as_bytes()
                    } else if task.is_terminated() {
                        "Terminated".as_bytes()
                    } else {
                        "Blocked".as_bytes()
```

Reply:
Alright, thanks. I added this hunk to the commit adding the BLOCKED task state.